pkg/virt-handler/node-labeller/node_labeller.go

```diff
@@ -0,0 +1,275 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package nodelabeller
+
+import (
+	"context"
+	"encoding/json"
+	"reflect"
+	"strings"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	libvirt "libvirt.org/libvirt-go"
+
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/client-go/log"
+	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
+	device_manager "kubevirt.io/kubevirt/pkg/virt-handler/device-manager"
+)
+
+const (
+	oldLabelNamespace              = "feature.node.kubernetes.io"
+	oldLabellerNamespaceAnnotation = "node-labeller-feature.node.kubernetes.io"
+	kvmPath                        = "/dev/kvm"
+	oldCPUFeaturePrefix            = "/cpu-feature-"
+	oldCPUModelPrefix              = "/cpu-model-"
+	oldHyperPrefix                 = "/kvm-info-cap-hyperv-"
+	CPUFeaturePrefix               = "cpu-feature.node.kubevirt.io/"
+	CPUModelPrefix                 = "cpu-model.node.kubevirt.io/"
+	HypervPrefix                   = "hyperv.node.kubevirt.io/"
+)
+
+//NodeLabeller struct holds informations needed to run node-labeller
+type NodeLabeller struct {
+	kvmController     *device_manager.DeviceController
+	nodeInformer      cache.SharedIndexInformer
+	clientset         kubecli.KubevirtClient
+	queue             workqueue.RateLimitingInterface
+	host              string
+	namespace         string
+	logger            *log.FilteredLogger
+	clusterConfig     *virtconfig.ClusterConfig
+	hypervFeatures    supportedFeatures
+	capScanner        *CapScan
+	hostCapabilities  supportedFeatures
+	cpuInfo           cpuInfo
+	virshCommandFN    func(*libvirt.Connect) (HostDomCapabilities, error)
```

Comment:
I wonder why does this has to part of this struct? 
Is it only for testing purposes? 
In that case, maybe it's better to create an interface and it will be easy to create a mock for it.
I see this function doesn't use anything from the NodeLabeller struct, so it shouldn't be a problem.

You could simply create a 

type LibvirtWrap interface {
    GetDomainCapabilities(*libvirt.Connect) (HostDomCapabilities, error)
}

and add a
`//go:generate mockgen -source $GOFILE -package=$GOPACKAGE -destination=generated_mock_$GOFILE`

on top of the file.
Perhaps, see[1] as a reference 

[1] https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-handler/device-manager/common.go#L37